### DISCUSSION TITLE
.NET 8 Support?

### DISCUSSION THREAD

**Discussion by Jukser** [Blazor-Diagrams/Blazor.Diagrams#387] [Q&A]
Hello
I'm trying to get Blazor-Diagrams work with .NET8  using new Blazor web App template.
I have to admit that I'm a bit newbie with blazor. I've played a little with .NET7 and Wasm and got Diagrams work earlier with .NET7. 
Now I'm trying to get the "getting started" sample work with .NET8 but so far no luck.
Page just shows text
Node1
Node2

Any tips how get things work with .NET8?

Cheers
Juha

here's code of MyDiagram.razor, that's in Components/pages folder in main project (not client):

@page "/MyDiagram"

@using Blazor.Diagrams.Components
@using Blazor.Diagrams.Core
@using Blazor.Diagrams
@using Blazor.Diagrams.Core.Anchors
@using Blazor.Diagrams.Core.Geometry
@using Blazor.Diagrams.Core.Models
@using Blazor.Diagrams.Core.PathGenerators
@using Blazor.Diagrams.Core.Routers
@using Blazor.Diagrams.Options
@using Microsoft.AspNetCore.Components

<div class="diagram-container">
    <CascadingValue Value="TestDiagram" IsFixed="true">
        <DiagramCanvas></DiagramCanvas>
    </CascadingValue>
</div>

@code{

    public BlazorDiagram TestDiagram { get; set; } = null!;

    protected override void OnInitialized()
    {
        base.OnInitialized();

        var options = new BlazorDiagramOptions
            {
                AllowMultiSelection = true,
                Zoom =
            {
                Enabled = false,
            },
                Links =
            {
                DefaultRouter = new NormalRouter(),
                DefaultPathGenerator = new SmoothPathGenerator()
            },
            };

        TestDiagram = new BlazorDiagram(options);
        var firstNode = TestDiagram.Nodes.Add(new NodeModel(position: new Point(50, 50))
            {
                Title = "Node 1"
            });
        var secondNode = TestDiagram.Nodes.Add(new NodeModel(position: new Point(200, 100))
            {
                Title = "Node 2"
            });
        var leftPort = secondNode.AddPort(PortAlignment.Left);
        var rightPort = secondNode.AddPort(PortAlignment.Right);

        // The connection point will be the intersection of
        // a line going from the target to the center of the source
        var sourceAnchor = new ShapeIntersectionAnchor(firstNode);
        // The connection point will be the port's position
        var targetAnchor = new SinglePortAnchor(leftPort);
        var link = TestDiagram.Links.Add(new LinkModel(sourceAnchor, targetAnchor));
    }
}

**Comment by zHaytam** [comment]
Hello, of course I'm planning to support .NET 8 I just need to find some time to do it.

**Comment by MindScriptAct** [comment]
I managed to make nodes appear, but links and interactions still not working.

**Reply by Jukser** [reply]
I got also nodes appear on Server side project., 
I'm not sure what I did differently I just followed the example and now it works. 
Maybe I missed some directions from App.razor. (ones you used to put Pages/_Host.cshtml)

I can move boxes and create links on sample.
Heading next to EventCallBack/Actions to capture my new links.

Anyways, @zHaytam  you have great library here.

**Comment by zHaytam** [comment]
A new version should be out soon that supports .net 8 thanks to a PR

**Reply by ATEEKGIT** [reply]
i don't know what needs to be changed for dotnet 8 , as seems like it works fine there as well. for webassembly and server and automode too.

**Reply by Sajith-TalentzQ** [reply]
Is .Net 8 supported for this ?

**Comment by fatihbahceci** [comment]
I tried it in .net 8 too. I got a blank screen. When I inspect in the browser, the drawings look transparent.

**Reply by ATEEKGIT** [reply]
we were using mudblazor and blazordiagrams , had more challanges to create the blazordiagrams inside a Mudblazor Dialog , after some tweaks it works just perfectly. in case you needs to see a working example of blazor diagrams  on  .Net8 with Mudlbaozr in operation. u may check this video.

https://www.youtube.com/watch?v=p61b8sXK52w

all i can say is you need to just dig in it to see where u have issue.

**Comment by fkmbkk2** [comment]
Hi Anyone manage to run this in Blazor .NET 8 ? Mine seems NOT working...

**Reply by liquick-net** [reply]
I am also struggling with getting this library working in .net 8. Maybe the first question that may shed light on my situation: can you use this library using the project template Blazor Web App --> Interactive render mode: server? Or do you need to create it as a WebAssembly?

**Comment by zHaytam** [comment]
For those who don't seem to get it working for .NET 8, what render mode are you using?

**Reply by krajurkarprovance** [reply]
Hi, I am also facing same problem , I am using VS2022, .NET8, project template Blazor Web App --> Interactive render mode: server.
When I install NuGet package  Z.Blazor.Diagrams it still says assembly reference missing. looks like it still doesn't support.
Has anyone found the solution?
